core/src/main/java/tc/oc/pgm/match/MatchImpl.java

```diff
@@ -811,6 +812,11 @@ public void load() throws ModuleLoadException {
       startTickables(MatchScope.LOADED);
       addParty(observers);
 
+      // This needs to be called BEFORE the match has completed loading
+      // but AFTER every other match module is finished with loading.
+      // This is to ensure that all match modules has registered their dynamic filters.
+      this.needModule(FilterMatchModule.class).onMatchLoad();
```

Comment:
this can not exist here.

You must find a way to make it compatible with simply using MatchLoadEvent on lowest or something along the lines
Why can this not happen before loaded is set to true?

Reply:
The old PGM dynamic system would prevent registering dynamic filters after the match started. Except for general security/avoiding bugs I can't imagine why this would be enforced

Reply:
Filters not being modifyiable after match start makes sense, but we probably want to allow them while match is still loading

Worst case scenario make "loaded" a 3-step enum so instead of false and true, you go from UNLOADED to LOADING, then match load event, and after the event you set it to LOADED

for the boolean-based "isLoaded"  check you return false for UNLOADED, and true for LOADING and LOADED (avoid breaking other things in the codebase, keeping the same behavior as now)
Then add another method that returns the actual value of the enum, and use that in the dynamic filter stuff, where you allow them being created in either UNLOADED or LOADING, but not LOADED

Reply:
Then, what about just allowing registering until `MatchScope.RUNNING`?